adb/adb_commands.py

```diff
@@ -151,6 +151,11 @@ def Push(self, source_file, device_filename, mtime='0', timeout_ms=None):
       mtime: Optional, modification time to set on the file.
       timeout_ms: Expected timeout for any part of the push.
     """
+
+    if os.path.isdir(source_file):
+      for dir in os.listdir(source_file):
+        return self.Push(os.path.join(source_file, dir), device_filename + "/" + dir)
```

Comment:
You're only returning the first one. You have to Push all of them then return after the for loop